src/Form/Control/Upload.php

```diff
@@ -161,54 +161,65 @@ public function addJsAction($action)
     }
 
     /**
-     * Call when user is removing an already upload file.
+     * Call when user is uploading a file.
      */
-    public function onDelete(\Closure $fx)
+    public function onUpload(\Closure $fx)
     {
-        $this->hasDeleteCb = true;
-        $action = $_POST['action'] ?? null;
-        if ($this->cb->triggered() && $action === 'delete') {
-            $this->_isCbRunning = true;
-            $fileName = $_POST['f_name'] ?? null;
-            $this->cb->set(function () use ($fx, $fileName) {
-                $this->addJsAction($fx($fileName));
+        $this->hasUploadCb = true;
```

Comment:
$this->hasUploadFx ?
But only if we can refactor $this->cb to $this->fx (BC)
To keep all naming consistent

Reply:
it is related with Callback, lets fix this in https://github.com/atk4/ui/pull/1377